rts/Rml/Backends/RmlUi_Backend.cpp

```diff
@@ -0,0 +1,441 @@
+/* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
+
+/*
+ * This source file is derived from the source code of RmlUi, the HTML/CSS
+ * Interface Middleware
+ *
+ * For the latest information, see http://github.com/mikke89/RmlUi
+ *
+ * Copyright (c) 2008-2010 CodePoint Ltd, Shift Technology Ltd
+ * Copyright (c) 2019-2023 The RmlUi Team, and contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+#include <RmlUi/Core.h>
+#include <RmlUi/Core/Profiling.h>
+#include <RmlUi/Debugger.h>
+#include <SDL.h>
+#include <functional>
+#include <tracy/Tracy.hpp>
+
+#include "Rendering/Textures/Bitmap.h"
+#include "Rml/Components/ElementLuaTexture.h"
+#include "Rml/RmlInputReceiver.h"
+#include "RmlUi_Backend.h"
+#include "RmlUi_Renderer_GL3_Recoil.h"
+#include "RmlUi_SystemInterface.h"
+#include "RmlUi_VFSFileInterface.h"
+#include "System/Input/InputHandler.h"
+#include "System/Log/ILog.h"
+
+using CtxMutex = std::recursive_mutex;
+using CtxLockGuard = std::lock_guard<CtxMutex>;
```

Comment:
Why do we need a mutex here?

Reply:
mostly to make sure that a lua script removing a context at an arbitrary point does not cause a crash